src/statusicon.cpp

```diff
@@ -67,38 +72,41 @@ void StatusIcon::activateAnimation() {
 
 void StatusIcon::animateIcon() {
   Q_ASSERT(m_animatedIconIndex < ANIMATED_LOGO_STEPS.size());
-  setIcon(ANIMATED_LOGO_STEPS[m_animatedIconIndex++], false);
+  setIcon(ANIMATED_LOGO_STEPS[m_animatedIconIndex++]);
   if (m_animatedIconIndex == ANIMATED_LOGO_STEPS.size()) {
     m_animatedIconIndex = 0;
   }
 }
 
-void StatusIcon::stabilityChanged() {
-  logger.debug() << "Stability changed";
+const QColor& StatusIcon::indicatorColor() {
+  logger.debug() << "Set color";
+
+  MozillaVPN* vpn = MozillaVPN::instance();
+  Q_ASSERT(vpn);
+
+  if (vpn->state() != MozillaVPN::StateMain ||
+      vpn->controller()->state() != Controller::StateOn) {
+    m_indicatorColor = QColor();
+    return m_indicatorColor;
+  }
 
-  switch (MozillaVPN::instance()->connectionHealth()->stability()) {
+  switch (vpn->connectionHealth()->stability()) {
     case ConnectionHealth::Stable:
-      setIndicatorColor(GREEN_COLOR);
+      m_indicatorColor = GREEN_COLOR;
       break;
     case ConnectionHealth::Unstable:
-      setIndicatorColor(ORANGE_COLOR);
+      m_indicatorColor = ORANGE_COLOR;
       break;
     case ConnectionHealth::NoSignal:
-      setIndicatorColor(RED_COLOR);
+      m_indicatorColor = RED_COLOR;
       break;
     default:
       logger.error() << "Unhandled status indicator for connection stability";
-      setIndicatorColor(QColor());
-  }
-}
-
-const QIcon& StatusIcon::icon() {
-  if (m_icon.isNull()) {
-    stateChanged();
-    Q_ASSERT(!m_icon.isNull());
+      m_indicatorColor = QColor();
+      break;
   }
 
-  return m_icon;
+  return m_indicatorColor;
 }
 
 void StatusIcon::stateChanged() {
```

Comment:
Here you can simply invalidate the m_icon and emit the signal. 
If you do this, you can remove `setIcon`